core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1170,8 +1170,8 @@ private JoinOnLeftCalcToJoinUnifyRule() {
       final RexBuilder rexBuilder = call.getCluster().getRexBuilder();
 
       // Try pulling up MutableCalc only when:
-      // 1. it's inner join;
-      // 2. it's outer join but no filttering condition from MutableCalc.
+      // 1. it's inner join.
+      // 2. it's outer join but no filtering condition from MutableCalc.
       final JoinRelType joinRelType = sameJoinType(query.joinType, target.joinType);
```

Comment:
This patch is small. Maybe we can merge it with others.

Reply:
+1 to merge it with other paths.